packages/core/src/crypto/JwkTypes.ts

```diff
@@ -1,6 +1,157 @@
+import { KeyType } from './KeyType'
+
 export interface Jwk {
   kty: 'EC' | 'OKP'
-  crv: 'Ed25519' | 'X25519' | 'P-256' | 'P-384' | 'secp256k1'
+  crv: 'Ed25519' | 'X25519' | 'P-256' | 'P-384' | 'P-521' | 'Bls12381G1' | 'Bls12381G2'
   x: string
   y?: string
+  use?: 'sig' | 'enc'
```

Comment:
What do we do if the `use` param isn't provided? According to the spec we can enforce it:


[4.2](https://www.rfc-editor.org/rfc/rfc7517#section-4.2).  "use" (Public Key Use) Parameter

   The "use" (public key use) parameter identifies the intended use of
   the public key.  The "use" parameter is employed to indicate whether
   a public key is used for encrypting data or verifying the signature
   on data.

   Values defined by this specification are:

   o  "sig" (signature)
   o  "enc" (encryption)

   Other values MAY be used.  The "use" value is a case-sensitive
   string.  Use of the "use" member is OPTIONAL, **unless the application
   requires its presence.**

Reply:
I think this depends on how purely we want to follow the spec. If we go 100%, we need to allow arbitrary string values as well. If not, I think we should maybe enforce the user to specify its value?

Reply:
For now, we just use it in parsing. It's required to be understood (not required to be present) for did:jwk. So we use when handling jwks, but only when we parse a JWK from someone else. When we create a JWK, for now I just always use the full capabilities of a key, and we don't include the `use` key. That's to to keep it 'simpler' (support the spec, but don't expose all functionality on our side).

I can update it to be a 'enc' | 'sig' | string value

Reply:
What do you mean by "required to be understood"?

I'm fine with keeping it as is as long as we actually use it. Adding the `string` type without adding logic to handle that doesn't make sense to me.

I was just wondering why we include it, as it's optional, only used for incoming JWKs, and only used by the `forEncrypting` and `forSigning` getters. Can we actually use those getters (as the `use` value is optional)?

Reply:
> I was just wondering why we include it

Because it's required by the `did:jwk` spec:

> If the JWK contains a use property with the value "sig" then the keyAgreement property is not included in the DID Document. If the use value is "enc" then only the keyAgreement property is included in the DID Document.
> The JWK should have the appropriate use value set to match the capabilities of the specified crv. For example, the 
curve ed25519 is only valid for "sig" use and X25519 is only valid for "enc" (see [RFC 8037](https://datatracker.ietf.org/doc/html/rfc8037) and the second example below).
> The JWK may contain additional custom properties and values which will be accessible only in the verificationMethod. Any additional properties other than use (as documented above) are not referenced or used in the generation of the DID Document.


So we use it in the minimal way possible. When we parse a `did:jwk` we do the following:
1. If no `use` property is present, we check the default capabilities of a key type (enc, sig or both)
2. If a `use` property is present, we check if the value is valid for the key type (ed25519 can't have `enc`). If it's valid we only inlcude the verificadtionMethod types that link to sig vs enc. So if it's enc, only the keyAgreement will be added to the resolved did:jwk document